test/unit/packwerk/package_todo_test.rb

```diff
@@ -224,6 +224,32 @@ class PackageTodoTest < Minitest::Test
       end
     end
 
+    test "#dump dumps package TODO comment" do
+      Tempfile.create("test_file.yml") do |file|
+        package_todo = PackageTodo.new(destination_package, file.path)
+        first_package = Package.new(name: "some_package", config: {})
+        first_package_reference = build_reference(
+          destination_package: first_package,
+          constant_name: "::Checkout::Wallet",
+          path: "some/violated/path.rb"
+        )
+
+        package_todo.add_entries(first_package_reference,
+          ReferenceChecking::Checkers::DependencyChecker::VIOLATION_TYPE)
+        package_todo.dump
+
+        assert_equal(<<~YAML, file.readlines.first(7).join)
+          # This file contains a list of dependencies that are not part of the long term plan for buyers.
```

Comment:
I've put up a PR to follow up on that thought https://github.com/Shopify/packwerk/pull/323

Reply:
As for the original topic of this thread (sorry for the derailment), it seems that the string is from the `destination_package` as implemented at the bottom of the file and used at the top. It's not very obvious because of the size of the test file.

Reply:
Yes, it is from the destination package (see the method in the test case), which would be the one with the violations. It is just confusing in the test because we use tempfiles and mock the setup a bit.